### PR TITLE
chore: split save resource asset

### PR DESCRIPTION
**Pull request by athurg**
This PR split the code for resource blob save into a function. After that, all resource upload use this function to save blob, include:

- Upload from HTTP API `POST /resource/blob`
- Upload from HTTP API `POST /resource` with external URL link.
- Upload from Telegram message

### CODE REVIEW

File: api/v1/resource.go [hunk 1]
```diff
@@ -659,3 +556,116 @@ func convertResourceFromStore(resource *store.Resource) *Resource {
 		LinkedMemoAmount: resource.LinkedMemoAmount,
 	}
 }
+
+// SaveResourceBlob save the blob of resource based on the storage config
+//
+// Depend on the storage config, some fields of *store.ResourceCreate will be changed:
+// 1. *DatabaseStorage*: `create.Blob`.
+// 2. *LocalStorage*: `create.InternalPath`.
+// 3. Others( extern service): `create.ExternalLink`.
```
Comment (boojack) [inline api/v1/resource.go:565]:
```suggestion
// 3. Others(external service): `create.ExternalLink`.
```

File: api/v1/resource.go [hunk 2]
```diff
@@ -659,3 +556,116 @@ func convertResourceFromStore(resource *store.Resource) *Resource {
...
+		_, err = io.Copy(dst, r)
+		if err != nil {
+			return fmt.Errorf("Failed to copy file: %s", err)
+		}
+
+		create.InternalPath = filePath
+		return nil
+	}
+
+	// Others: store blob into extern service, such as S3
```
Comment (boojack) [inline api/v1/resource.go:627]:
```suggestion
	// Others: store blob into external service, such as S3
```

**Review by boojack — APPROVED** [review]
LGTM, otherwise

**Comment by athurg** [comment]
Wait a minute, I'll rebase on the latest `main` branch to fix the conflict.

**Comment by athurg** [comment]
It's ready now, you can just merge this if there's any other problems.

**Pull request merged**